### PR TITLE
Fixes issue in the outbound local mx check introduced in #3002

### LINKED ISSUE #2809: Localhost mail loop problem
I am trying to solve kind of configuration problem - it is not Haraka issue per se

There are multiple servers which returns MX as 0.0.0.0 or 127.0.0.1, or other "problematic" IPs. What happens is that Haraka will deliver to itself and relay via outbound again. It depends on added headers but after ±100 deliveries Haraka will reject mail as looped mail. 

Thing is that malicious user can easily induce dos with this kind of behaviour. Only solution I've come up is kind of hacky with abusing metadata in headers. Am I missing something?

### PR DESCRIPTION
**Pull request by DoobleD**
Fixes https://github.com/haraka/Haraka/pull/3002#issuecomment-1002750946.

Changes proposed in this pull request:
- make `process_dns` async
- use `await` in front of `is_local_host`

Requires https://github.com/haraka/haraka-net-utils/pull/65 to be merged, as that PR refactors `is_local_host` to use promises.

Alternatively it should be possible to use callbacks only, but it seems to me the code is much simpler using promises.

Checklist:
- [ ] docs updated
- [ ] tests updated
- [ ] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

File: outbound/mx_lookup.js [hunk 1]
```diff
@@ -35,7 +35,10 @@ exports.lookup_mx = function lookup_mx (domain, cb) {
         }
         else if (addresses && addresses.length) {
             for (let i=0,l=addresses.length; i < l; i++) {
-                if (obc.cfg.local_mx_ok || !net_utils.is_local_ip(addresses[i].exchange)) {
+                if (
+                    obc.cfg.local_mx_ok ||
+                    await net_utils.is_local_host(addresses[i].exchange).catch(() => null) === false
```
Comment (DoobleD) [inline outbound/mx_lookup.js:40]: The `catch` is just to be extra safe and ignore a potential DNS resolution failure.

**Comment by DoobleD** [comment]
> This pull request **introduces 4 alerts** when merging [ea9398f](https://github.com/haraka/Haraka/commit/ea9398fec4aa5c76067f6c13c591adf0fb089e8c) into [786daac](https://github.com/haraka/Haraka/commit/786daac7328399efb29f124862801da3edf3e3b5) - [view on LGTM.com](https://lgtm.com/projects/g/haraka/Haraka/rev/pr-a511cbd9e3ad57beb0e4ce69f592b95e1a7e01f8)
> 
> **new alerts:**
> 
> * 2 for Useless conditional
> * 2 for Missing await

Fixed.

File: package.json [hunk 2]
```diff
@@ -20,61 +20,61 @@
     "node": ">=12.22.6"
   },
   "dependencies": {
-    "address-rfc2821"       : "^2.0.1",
```
Comment (DoobleD) [inline package.json:23]: Those formatting changes are not intended, perhaps the linter or my IDE did it, I'm not sure.

File: package.json [hunk 3]
```diff
@@ -20,61 +20,61 @@
...
-    "haraka-dsn"            : "^1.0.3",
-    "haraka-net-utils"      : "^1.3.2",
-    "haraka-notes"          : "^1.0.4",
-    "haraka-plugin-redis"   : "^1.0.12",
-    "haraka-results"        : "^2.0.3",
-    "haraka-tld"            : "^1.0.28",
-    "haraka-utils"          : "^1.0.2",
-    "openssl-wrapper"       : "^0.3.4",
-    "sockaddr"              : "^1.0.1"
+    "address-rfc2821": "^2.0.1",
```
Comment (DoobleD) [inline package.json:23]: Those formatting changes are not intended, perhaps the linter or my IDE did it, I'm not sure.

File: package.json (hunk 2 above)
Comment (DoobleD) [inline package.json:23]: Those formatting changes are not intended, perhaps the linter or my IDE did it, I'm not sure.

Comment (msimerson) [reply package.json:23]: Please keep the previous formatting.

Comment (DoobleD) [reply package.json]: Done!

**Review by msimerson — APPROVED** [review]

**Pull request merged**